webapp/store/views.py

```diff
@@ -62,18 +62,19 @@ def store_view():
 
         try:
             categories_results = api.get_categories()
-        except ApiError as api_error:
+        except ApiError:
             categories_results = []
-            status_code, error_info = _handle_errors(api_error)
 
         categories = logic.get_categories(categories_results)
 
         try:
             featured_snaps_results = api.get_searched_snaps(
                 snap_searched="", category="featured", size=10, page=1
             )
-        except ApiError:
+        except ApiError as api_error:
             featured_snaps_results = []
+            status_code, error_info = _handle_errors(api_error)
+            flask.abort(status_code)
```

Comment:
Should it be `return flask.abort` (at least it seems what we are doing elsewhere).
```suggestion
            return flask.abort(status_code)
```